### PR TITLE
Replace Search directory by the ArborX submodule

### PR DESCRIPTION
**Pull request by aprokop**
Other steps to do (on top of my head):
- [x] Update ReadTheDocs instructions
- [ ] Update DTK's Spack (in progress)
  Need to clone submodules too
- [x] Update build scripts
  Don't run benchmark as we don't compile ArborX tests in DTK anymore
- [ ] ~Update Docker (optional)~
  No longer need to install benchmark (and maybe something else)

### CODE REVIEW

**Comment by aprokop** [comment]
Some of the recent updates in ArborX make this fail. Specifically, the addition of `ArborX_Config.hpp` which is *generated* by ArborX during configuration, but is not part of the repo. Thus, the DTK build fails as it cannot find it. As MPI is a required dependency in DTK, and (now) an optional in ArborX, we need to have `ArborX_Config.hpp` with `#define ARBORX_ENABLE_MPI` in it.

This is more than we originally as it requires more than just grabbing headers in DTK.

**Comment by aprokop** [comment]
Getting
```
/scratch/source/trilinos/release/DataTransferKit/packages/Search/src/details/ArborX_DetailsUtils.hpp:172:3: error: no template named 'MinMaxScalar' in namespace 'Kokkos'; did you mean 'Kokkos::Experimental::MinMaxScalar'?
```
It seems that we need to update the Kokkos version for DTK testing to match the one for ArborX.

**Comment by dalg24** [comment]
Deprecated code is disabled when building Kokkos for ArborX.

**Comment by aprokop** [comment]
retest this please

**Comment by aprokop** [comment]
retest this please

**Comment by aprokop** [comment]
This is ready for review, btw.

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,6 @@
 [submodule "data"]
 	path = data
 	url = https://github.com/ORNL-CEES/DTKData.git
+[submodule "arborx"]
+	path = packages/Search
+	url = https://github.com/arborx/arborx.git
```
Comment (dalg24) [inline .gitmodules]: Not case sensitive?

File: docs/source/install.rst [hunk 2]
```diff
@@ -19,8 +19,6 @@ The following third party libraries (TPLs) are used by DTK:
 +------------------------+-------------------------------------+---------+
 | Trilinos               | Required                            | 12.X    |
 +------------------------+-------------------------------------+---------+
-| Google Benchmark       | Required (if examples are enabled)  | 1.4     |
-+------------------------+-------------------------------------+---------+
```
Comment (dalg24) [inline docs/source/install.rst]: That's good news

File: .gitmodules (hunk 1 above)
Comment (aprokop) [reply .gitmodules]: http url are not case sensitive.

Comment (Rombur) [reply .gitmodules]: There are though. Only the domain name is not case sensitive https://stackoverflow.com/questions/7996919/should-url-be-case-sensitive

Comment (dalg24) [reply .gitmodules]: FWIW I did a clone recursive of Andrey's branch and it worked

Comment (aprokop) [reply .gitmodules]: @Rombur Right you are.

That being said, github *is* case-insensitive for repos and organizations (though I have not found the exact place on GitHub stating that), so it's not a big deal.

**Comment by aprokop** [comment]
@Rombur @dalg24 Do you need me to change the URL, or are you OK with the way it is?

**Comment by Rombur** [comment]
If it works, it's fine with me.

**Pull request merged**